### PR TITLE
Provide option to version info in generated .sys file

### PR DESCRIPTION
**Pull request by Alan-Jowett**
## Description

Provide option to Convert-BpfToNative.ps1 to pass resource file for use when generating .sys file.

## Testing

CI/CD

## Documentation

Added documentation to the script.

### CODE REVIEW

**Comment by gtrevi** [comment]
Should this PR also include the calls to actually add the version info in the generated sys files (i.e. for the next release), or is it planned in another PR?

File: tools/bpf2c/Convert-BpfToNative.ps1.template [hunk 1]
```diff
@@ -1,6 +1,73 @@
 # Copyright (c) Microsoft Corporation
 # SPDX-License-Identifier: MIT
 
+<#
+.SYNOPSIS
+    Generate a native driver from a BPF program.
+
+.DESCRIPTION
+    This script generates a native driver (.sys file) from a BPF program. The native program can be used to load BPF
+    programs into the kernel when Hyper-V Code Integrity (HVCI) is enabled.
```
Comment (dthaler) [inline tools/bpf2c/Convert-BpfToNative.ps1.template:10]:
```suggestion
    programs into the kernel when Hypervisor-protected Code Integrity (HVCI) is enabled.
```
fix expansion of acronym

File: tools/bpf2c/Convert-BpfToNative.ps1.template [hunk 2]
```diff
@@ -1,6 +1,73 @@
...
+.PARAMETER KernelMode
+    Specifies whether to generate a kernel-mode driver. If this parameter is false, a user DLL will be generated.
+
+.PARAMETER ResourceFile
+    Specifies the path to a resource file to embed in the generated driver.
+
+.EXAMPLE
+    .\Convert-BpfToNative.ps1 -ProgramName bindmonitor
+
+    This example generates a native driver from the BPF program bindmonitor.o.
```
Comment (dthaler) [inline tools/bpf2c/Convert-BpfToNative.ps1.template:45]: Don't confuse the filename with the program name.   "bindmonitor.o" is a filename and may be completely different from the program name.  (Same problem in lines 50, 13, etc.)

**Comment by dthaler** [comment]
> > Should this PR also include the calls to actually add the version info in the generated sys files (i.e. for the next release), or is it planned in another PR?
> 
> Which version number should be included? There is no version number in the input ELF file, so this PR permits the end user to add their own version number / copyright etc into the resulting .sys file.

I'd say we should start using the standard version number capability in the input ELF file.
https://github.com/dthaler/ebpf-docs-1/blob/elf/rst/elf.rst#runtime-version-restriction

**Comment by gtrevi** [comment]
> > Should this PR also include the calls to actually add the version info in the generated sys files (i.e. for the next release), or is it planned in another PR?
> 
> Which version number should be included? There is no version number in the input ELF file, so this PR permits the end user to add their own version number / copyright etc into the resulting .sys file.

As discussed offline, agree to leave the decision to the customers, to provide their own resource file, accordingly to their versioning strategy.

**Review by gtrevi — APPROVED** [review]

File: tools/bpf2c/Convert-BpfToNative.ps1.template (hunk 2 above)
Comment (Alan-Jowett) [reply tools/bpf2c/Convert-BpfToNative.ps1.template:45]: OK, will switch it to filename.

**Review by gtrevi — APPROVED** [review]

**Review by shankarseal — APPROVED** [review]

**Pull request merged**